File: packages/python.org/libraries/lodash/tests/string.test.py
"""
string
"""

import unittest
import libs as _


class TestStringMethods(unittest.TestCase):
    def test_camelCase(self):
        self.assertEqual(_.camel_case("Foo bar"), "fooBar")

    def test_capitalize(self):
        self.assertEqual(_.capitalize("FOO"), "Foo")

    def test_ends_with(self):
        self.assertTrue(_.ends_with("abc", "c"))
        self.assertFalse(_.ends_with("abc", "b"))
        self.assertTrue(_.ends_with("abc", "b", 2))

    def test_kebabCase(self):
        self.assertEqual(_.kebab_case("Foo Bar"), "foo-bar")

    def test_lowerCase(self):
        self.assertEqual(_.lower_case("--Foo--Bar--"), "foo bar")
        self.assertEqual(_.lower_case("__Foo__Bar__"), "foo bar")

    def test_lowerFirst(self):
        self.assertEqual(_.lower_first("Fred"), "fred")
        self.assertEqual(_.lower_first("F_rED"), "f_rED")

    def test_pad(self):
        self.assertEqual(_.pad("abc", 8), "    abc     ")
        self.assertEqual(_.pad("abc", 8, "_-"), "_-abc_-_")
        self.assertEqual(_.pad("abc", 3), "abc")

    def test_padEnd(self):
        self.assertEqual(_.pad_end("abc", 6), "abc     ")
        self.assertEqual(_.pad_end("abc", 6, "_-"), "abc_-_")
        self.assertEqual(_.pad_end("abc", 3), "abc")

    def test_padStart(self):
        self.assertEqual(_.pad_start("abc", 6), "     abc")
        self.assertEqual(_.pad_start("abc", 6, "_-"), "_-_abc")
        self.assertEqual(_.pad_start("abc", 3), "abc")

    def test_parseInt(self):
        self.assertEqual(_.parse_int("08"), 8)

    def test_repeat(self):
        self.assertEqual(_.repeat("*", 3), "***")

    def test_replace(self):
        self.assertEqual(_.replace("placeholder", "placeholder", "test"), "test")

    def test_snakeCase(self):
        self.assertEqual(_.snake_case("Foo Bar"), "foo_bar")

    def test_split(self):
        self.assertEqual(_.split("a-b-c", "-"), ["a", "b", "c"])
        self.assertEqual(_.split("a-b-c", "-", 2), ["a", "b"])

    def test_startCase(self):
        self.assertEqual(_.start_case("--foo--bar--"), "Foo Bar")
        self.assertEqual(_.start_case("__foo__bar__"), "Foo Bar")

    def test_starts_with(self):
        self.assertTrue(_.starts_with("test", "t"))
        self.assertFalse(_.starts_with("test", "e"))
        self.assertTrue(_.starts_with("test", "e", 1))

    def test_toLower(self):
        self.assertEqual(_.to_lower("TEST"), "test")

    def test_toUpper(self):
        self.assertEqual(_.to_upper("test"), "TEST")

    def test_trim(self):
        self.assertEqual(_.trim("     abc     "), "abc")

    def test_trimEnd(self):
        self.assertEqual(_.trim_end("     abc     "), "     abc")

    def test_trimStart(self):
        self.assertEqual(_.trim_start("     abc     "), "abc     ")

    def test_truncate(self):
        self.assertEqual(_.truncate("hi-diddly-ho there, neighborino"), "hi-diddly-ho there, neighbo ...")

    def test_upperCase(self):
        self.assertEqual(_.upper_case("--Foo--Bar--"), "FOO BAR")
        self.assertEqual(_.upper_case("__Foo__Bar__"), "FOO BAR")

    def test_upper_first(self):
        self.assertEqual(_.upper_first("tEST"), "Test")

    def test_words(self):
        self.assertEqual(_.words("fred, barney, & pebbles"), ["fred", "barney", "pebbles"])


if __name__ == "__main__":
    unittest.main()

File: packages/go.dev/libraries/utils/lodash/string/string_test.go
package string

import (
	"fmt"
	"testing"
)

func TestEndsWith(t *testing.T) {
	var _endsWith1 bool = EndsWith("abc", 'c', 1)
	var _endsWith2 bool = EndsWith("abc", 'b', 1)
	var _endsWith3 bool = EndsWith("abc", 'b', 2)
	if _endsWith1 && !_endsWith2 && _endsWith3 {
		fmt.Println("endsWith true")
	}
}

func TestTrim(t *testing.T) {
	var _trim string = Trim("  abc  ", " ")
	fmt.Println("trim", _trim == "abc")
}

func TestTrimEnd(t *testing.T) {
	var _trimEnd string = TrimEnd("  abc  ", " ")
	fmt.Println("trimEnd", _trimEnd == "  abc")
}

func TestTrimStart(t *testing.T) {
	var _trimStart string = TrimStart("  abc  ", " ")
	fmt.Println("trimStart", _trimStart == "abc  ")
}

func TestStartsWith(t *testing.T) {
	var _startsWith1 bool = StartsWith("abc", 'a', 0)
	var _startsWith2 bool = StartsWith("abc", 'a', 1)
	var _startsWith3 bool = StartsWith("abc", 'b', 1)
	if _startsWith1 && !_startsWith2 && _startsWith3 {
		fmt.Println("startsWith true")
	}
}
